### PR TITLE
add enterprise-openedx-operator permission checks

### PR DESCRIPTION
**Pull request by mushtaqak**
Description: Add permission checks for enterprise_openedx_operator role and update the logic for existing implicit and explicit access checks.

[ENT-1737](https://openedx.atlassian.net/browse/ENT-1737)

### CODE REVIEW

**Comment by mushtaqak** [comment]
build is not being triggered, closing and re-opening ...

File: ecommerce/enterprise/rules.py [hunk 1]
```diff
@@ -12,20 +12,35 @@
...
 )
 
-from ecommerce.core.constants import ENTERPRISE_COUPON_ADMIN_ROLE
+from ecommerce.core.constants import ENTERPRISE_COUPON_ADMIN_ROLE, SYSTEM_ENTERPRISE_OPERATOR_ROLE
 from ecommerce.core.models import EcommerceFeatureRoleAssignment
 from ecommerce.enterprise.constants import USE_ROLE_BASED_ACCESS_CONTROL
 
 
+@rules.predicate
+def request_user_is_edx_operator(user, context):    # pylint: disable=unused-argument
```
Comment (brittneyexline) [inline ecommerce/enterprise/rules.py]: i think as long as the system wide role gets mapped to the coupon admin role in the settings, this check won't be needed. the context for the system wide role will be `*`, which is now supported to mean access to all enterprises based on Ammar's PR. so the existing check for the single feature specific role of enterprise_coupon_admin should be sufficient.

**Review by christopappas — APPROVED** [review]
Waffle switch refactor stuff looks good. I think we can remove the new predicate here because the same logic is covered in ammar's edx-rbac PR, other than that we're 👍

File: ecommerce/core/models.py [hunk 2]
```diff
@@ -753,9 +754,12 @@ class EcommerceFeatureRoleAssignment(UserRoleAssignment):
 
     def get_context(self):
         """
-        Return the context for this role assignment class.
+        Return the enterprise customer id or `*` if the user has access to all resources.
         """
-        return self.enterprise_id and str(self.enterprise_id)
+        enterprise_id = ALL_ACCESS_CONTEXT
+        if self.enterprise_id:
+            enterprise_id = self.enterprise_id and str(self.enterprise_id)
```
Comment (muhammad-ammar) [inline ecommerce/core/models.py]: I think we only need `enterprise_id = str(self.enterprise_id)`

**Review by muhammad-ammar — CHANGES_REQUESTED** [review]
LGTM There is one minor comment.

**Review by muhammad-ammar — APPROVED** [review]

**Comment by mushtaqak** [comment]
@brittneyexline Can you please merge this? :shipit:

**Pull request merged**